RecoEgamma/EgammaPhotonProducers/interface/PhotonProducer.h

```diff
@@ -28,8 +28,6 @@
 #include "RecoEgamma/PhotonIdentification/interface/PhotonIsolationCalculator.h"
 #include "RecoLocalCalo/EcalRecAlgos/interface/EcalSeverityLevelAlgo.h"
 #include "RecoEgamma/PhotonIdentification/interface/PhotonMIPHaloTagger.h"
-#include "RecoEcal/EgammaCoreTools/interface/EcalClusterFunctionFactory.h"
-#include "RecoEcal/EgammaCoreTools/interface/EcalClusterFunctionBaseClass.h"
```

Comment:
This is still needed by https://github.com/cms-sw/cmssw/blob/10c3304da8db43373f72e8bac80336796df148f7/RecoEgamma/EgammaPhotonProducers/interface/PhotonProducer.h#L103

On the other hand, any usage of such ```energyCorrectionF``` is commented out in the class implementation. Therefore, it is ok to remove the include, provided you also remove:
- the definition of ```energyCorrectionF``` in PhotonProducer.h
- all its references (now commented out) in PhotonProducer.cc

And, since you are cleaning, you could also get rid of:
- https://github.com/cms-sw/cmssw/blob/10c3304da8db43373f72e8bac80336796df148f7/RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc#L264
- https://github.com/cms-sw/cmssw/blob/10c3304da8db43373f72e8bac80336796df148f7/RecoEgamma/EgammaPhotonProducers/interface/PhotonProducer.h#L50
- https://github.com/cms-sw/cmssw/blob/10c3304da8db43373f72e8bac80336796df148f7/RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc#L213-L214
- https://github.com/cms-sw/cmssw/blob/10c3304da8db43373f72e8bac80336796df148f7/RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc#L226
- https://github.com/cms-sw/cmssw/blob/10c3304da8db43373f72e8bac80336796df148f7/RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc#L355
- ... and all useless commented out ```cout```'s in PhotonProducer.cc